integrations/nvidia/src/haystack_integrations/components/rankers/nvidia/ranker.py

```diff
@@ -0,0 +1,247 @@
+class NvidiaRanker:
+    """
+    A component for ranking documents using ranking models provided by
+    [NVIDIA NIMs](https://ai.nvidia.com).
+
+    Usage example:
+    ```python
+    from haystack_integrations.components.rankers.nvidia import NvidiaRanker
+
+    ranker = NvidiaRanker(
+        model="nvidia/nv-rerankqa-mistral-4b-v3",
+        api_key="your-api-key",
+    )
+    ranker.warm_up()
+
+    query = "What is the capital of Germany?"
+    documents = [
+        Document(content="Berlin is the capital of Germany."),
+        Document(content="The capital of Germany is Berlin."),
+        Document(content="Germany's capital is Berlin."),
+    ]
+
+    result = ranker.run(query, documents, top_k=2)
+    print(result["documents"])
+    ```
+    """
+
+    def __init__(
+        self,
+        model: Optional[str] = _DEFAULT_MODEL,
+        truncate: Optional[Literal["NONE", "END"]] = None,
+        api_url: Optional[str] = None,
+        api_key: Optional[Secret] = None,
+        top_k: int = 5,
+    ):
+        """
+        Create a NvidiaRanker component.
+
+        :param model:
+            Ranking model to use.
+        :param truncate:
+            Truncation strategy to use. Can be "NONE" or "END". Defaults to NIM's default.
+        :param api_key:
+            API key for the NVIDIA NIM.
+        :param api_url:
+            Custom API URL for the NVIDIA NIM.
+        :param top_k:
+            Number of documents to return.
+        """
+        if not isinstance(model, str):
+            msg = "Ranker expects the `model` parameter to be a string."
+            raise TypeError(msg)
+        if not isinstance(api_url, (str, type(None))):
+            msg = "Ranker expects the `api_url` parameter to be a string."
+            raise TypeError(msg)
+        if truncate is not None and truncate not in ["NONE", "END"]:
+            msg = "Invalid truncate value. Must be 'NONE' or 'END'."
+            raise ValueError(msg)
+        if not isinstance(top_k, int) or isinstance(top_k, bool):
+            msg = "Ranker expects the `top_k` parameter to be an integer."
+            raise TypeError(msg)
+
+        self._model = model
+        self._truncate = truncate
+        self._api_key = api_key
+        # if no api_url is provided, we're using a hosted model and can
+        #  - assume the default url will work, because there's only one model
+        #  - assume we won't call backend.models()
+        if api_url is not None:
+            self._api_url = url_validation(api_url, None, ["v1/ranking"])
+            self._endpoint = None  # we let backend.rank() handle the endpoint
+        else:
+            if model not in _MODEL_ENDPOINT_MAP:
+                msg = f"Model '{model}' is unknown. Please provide an api_url to access it."
+                raise ValueError(msg)
+            self._api_url = None  # we handle the endpoint
+            self._endpoint = _MODEL_ENDPOINT_MAP[self._model]
+            if api_key is None:
```

Comment:
You can remove this and initialised like in my suggestions

Reply:
there are two modes of operation -

 1. hosted - calls to api.nvidia.com, these require an api_key
 2. local - calls to some user known api_url, these do not require an api_key

i would like to only require the api_key in the first case. ok to keep this?

Reply:
ah ok, I see, then here it's enough to to `api_key = Secret.from_env_var("NVIDIA_API_KEY")` and then call the `.resolve_value()` in the run() method

Reply:
is it sufficient that resolve_value is called in the NimBackend init?

see https://github.com/deepset-ai/haystack-core-integrations/pull/1023/files#diff-4f9b82acfb7e3f7ebe10dc8f69f7f3e26a01b24e34c7b8a74cfd55c372cda5aaR41

Reply:
ok, it's called there, I missed that